File: spring-parent/spring-demoapp/src/main/java/com/example/springapp/web/MetadataDto.java
package com.example.springapp.web;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Getter
@Setter
public class MetadataDto {

    private Long movieId;
    private String title;
    private Integer year;
    private Long directorId;
    private String directorName;
    private List<String> directorOtherMovies;
    private int movieVersion;
    private String movieCreatedDate;
    private int directorVersion;
    private String directorCreatedDate;
    private String _self;

}

File: spring-demoapp-kotlin/src/main/java/com/example/springapp/web/MetadataDto.kt
package com.example.springapp.web

data class MetadataDto (
        var movieId: Long? = null,
        var title: String? = null,
        var year: Int? = null,
        var directorId: Long? = null,
        var directorName: String? = null,
        var directorOtherMovies: List<String>? = null,
        var movieVersion: Int? = null,
        var movieCreatedDate: String? = null,
        var directorVersion: Int? = null,
        var directorCreatedDate: String? = null,
        var _self: String? = null)